File: activity_05_jflex/src/main/java/Lexeme.java
public class Lexeme extends Symbol {

    private Token token;

    public Lexeme(final String label, final Token token) {
        super(label);
        this.token = token;
    }

    public Token getToken() {
        return token;
    }

    public void setToken(Token token) {
        this.token = token;
    }

    @Override
    public String toString() {
        return "(label:" + label + ", token:" + token + ")";
    }

    public static void main(String[] args) {
        Lexeme lexeme = new Lexeme("+", Token.ADDITION);
        System.out.println(lexeme);
    }
}

File: activity_07_topdown_parser/src/Lexeme.scala
/*
 * CS3210 - Principles of Programming Languages - Fall 2021
 * Instructor: Thyago Mota
 * Description: Activity 07 - Lexeme (a symbol with a token value)
 */

class Lexeme(private var label: String, private var token: Token.Value) extends Symbol(label) {

  def getToken() = { token }

  override def toString: String = "(label:" + label + ", token:" + token + ")"
}